pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,427 @@
+            del item
+        for chart_name, chart in self.charts.items():
+            self.disconnect_all(chart)
+
+    def makeLayout(self, axes=None, charts=None):
+        """Adds all given axes and charts to the widget.
+
+        Parameters
+        ----------
+        axes : list of str or None:
+            The names associated with the axes to show.
+        charts : list of PlotItems or None:
+            The names associated with the charts to show.
+        """
+        self.clearLayout()
+        shown_axes = self.show_axes(axes)
+        shown_charts = self.show_charts(charts)
+        if len(shown_charts) != 0:
+            top_level = shown_charts[list(shown_charts)[-1]]
+            # FROM "https://github.com/pyqtgraph/pyqtgraph/pull/2010" by herodotus77
+            # propagate mouse actions to charts "hidden" behind
+            for k, c in self.charts.items():
+                self.connect_signals(top_level, c)
+        # MOVE LEGEND TO LAYOUT
+        if self.pi.legend is not None:
+            self.pi.legend.setParentItem(self.pi)
+        self.update()
+
+    def show_axes(self, axes=None):
+        """Shows all the selected axes."""
+        # SELECT AND ASSEMBLE AXES
+        if axes is None:
+            axes = list(self.axes)
+        lo = {
+            "left": [],
+            "right": [],
+            "top": [],
+            "bottom": [],
+        }
+        for k, a in self.axes.items():
+            if k in axes:
+                lo[a.orientation].append(a)
+        vx = len(lo["left"])
+        vy = 1 + len(lo["top"])
+        # ADD TITLE ON TOP
+        self.pi.titleLabel.show()
+        self.layout.addItem(self.pi.titleLabel, 0, vx)
+        # ADD MAIN PLOTITEM
+        self.vb.show()
+        self.layout.addItem(self.vb, vy, vx)
+        # ADD AXIS
+        shown_axes = {"x": {}, "y": {}}
+        for x, a in enumerate(lo["left"] + [None] + lo["right"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, vy, x)
+                shown_axes["x"][a.name] = a
+        for y, a in enumerate([None] + lo["top"] + [None] + lo["bottom"]):
+            if a is not None:
+                a.show()
+                self.layout.addItem(a, y, vx)
+                shown_axes["y"][a.name] = a
+        return shown_axes
+
+    def show_charts(self, charts=None):
+        """Shows all the selected charts."""
+        # SELECT CHARTS
+        if charts is None:
+            charts = self.charts
+        shown_charts = {}
+        for k, c in self.charts.items():
+            if k in charts:
+                c.show()
+                shown_charts[k] = c
+            else:
+                c.hide()
+        return shown_charts
+
+    def connect_signals(self, top_level, chart):
+        """Connects all signals related to this widget for the given chart given the top level one."""
+        self.disconnect_all(chart)
+        tvb = top_level.plotItem.vb
```

Comment:
generally I try and avoid abbreviations like these as they can be a bit tough to track.  I recognize that some of these lines are already getting long; but I think doing an alias like `signals = self._signalConnectionsByChart[chart.name]` would go a long way to making it a bit easier to read.

Reply:
Improved shortcuts names